cobc/error.c

```diff
@@ -63,10 +70,73 @@ print_error_prefix (const char *file, int line, const char *prefix)
 	}
 }
 
+/* Display a context around the location of the error:
+   * only if cb_diagnostics_show_caret is true
+   * only if it is an error, or a warning and cb_diagnostics_warning_show_caret
+     is true
+   Only display two lines before and after. No caret yet for the column as
+   we only have the line. Since we directly use the file, source is printed
+   before any REPLACE.
+ */
+static void
+diagnostics_show_caret  (enum cb_error_kind kind, const char *file, int line)
+{
+	if ( cb_diagnostics_show_caret
+	     && file != NULL && line
+	     && ( cb_diagnostics_warning_show_caret || kind == CB_KIND_ERROR )
+		){
+		FILE* fd = fopen(file, "r");
```

Comment:
needs to check `fd` after open and likely should check before and not do that if name matches `COB_DASH`

Reply:
I added checks for NULL and COB_DASH.